app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -218,9 +216,17 @@ class Settings private constructor(
 
     var shouldUseBottomToolbar by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_key_toolbar_bottom),
-        default = true
+        // Default accessibility users to top toolbar
+        default = !touchExplorationIsEnabled
     )
 
+    private val touchExplorationIsEnabled: Boolean
+        get() {
+            val accessibilityManager =
+                appContext.getSystemService(Context.ACCESSIBILITY_SERVICE) as? AccessibilityManager
+            return accessibilityManager?.isTouchExplorationEnabled ?: false
```

Comment:
I don't think the switch service makes `isTouchExplorationEnabled == true`. Does it? You may need to sniff that out by name or find another useful attribute to detect with.

Reply:
I will have QA test with switch service. I don't see any documentation from Google for a flag that specifically mentions switch service so I'm not sure how we could detect that specifically (if it's not covered by `isTouchExplorationEnabled`)

Reply:
I wrote a little app that dumps the a11y service info, trying to see what kinds of flags/capabilities we should sniff out to enable "a11y mode" for the toolbar.

It looks like the switch service does not enable explore by touch.
It also looks like if a service has `CAPABILITY_CAN_PERFORM_GESTURES`, it may be a switch or switch-like service. I can't say that with full confidence that there are no other 3rd party a11y services out there that use that capability differently, and would not be happy to see the toolbar change.

```json
{
 "isTouchExplorationEnabled": false,
 "isEnabled": true,
 "services": [
  {
   "id": "com.google.android.marvin.talkback/.TalkBackService",
   "capabilities": [
    "CAPABILITY_CAN_RETRIEVE_WINDOW_CONTENT (0x1)",
    "CAPABILITY_CAN_REQUEST_TOUCH_EXPLORATION (0x2)",
    "CAPABILITY_CAN_REQUEST_FILTER_KEY_EVENTS (0x8)",
    "CAPABILITY_CAN_CONTROL_MAGNIFICATION (0x10)",
    "CAPABILITY_CAN_REQUEST_FINGERPRINT_GESTURES (0x40)"
   ],
   "flags": [
    "DEFAULT (0x1)",
    "FLAG_REQUEST_TOUCH_EXPLORATION_MODE (0x4)",
    "FLAG_REPORT_VIEW_IDS (0x10)",
    "FLAG_REQUEST_FILTER_KEY_EVENTS (0x20)",
    "FLAG_RETRIEVE_INTERACTIVE_WINDOWS (0x40)",
    "FLAG_ENABLE_ACCESSIBILITY_VOLUME (0x80)",
    "null (0x400)"
   ],
   "feedbackType": "[FEEDBACK_SPOKEN, FEEDBACK_HAPTIC, FEEDBACK_AUDIBLE]"
  },
  {
   "id": "com.google.android.marvin.talkback/com.google.android.accessibility.accessibilitymenu.AccessibilityMenuService",
   "capabilities": [
    "CAPABILITY_CAN_REQUEST_FILTER_KEY_EVENTS (0x8)"
   ],
   "flags": [
    "FLAG_REQUEST_FILTER_KEY_EVENTS (0x20)",
    "FLAG_REQUEST_ACCESSIBILITY_BUTTON (0x100)"
   ],
   "feedbackType": "[FEEDBACK_GENERIC]"
  },
  {
   "id": "com.google.android.marvin.talkback/com.google.android.accessibility.selecttospeak.SelectToSpeakService",
   "capabilities": [
    "CAPABILITY_CAN_RETRIEVE_WINDOW_CONTENT (0x1)",
    "CAPABILITY_CAN_REQUEST_FILTER_KEY_EVENTS (0x8)"
   ],
   "flags": [
    "DEFAULT (0x1)",
    "FLAG_REPORT_VIEW_IDS (0x10)",
    "FLAG_REQUEST_FILTER_KEY_EVENTS (0x20)",
    "FLAG_RETRIEVE_INTERACTIVE_WINDOWS (0x40)",
    "FLAG_ENABLE_ACCESSIBILITY_VOLUME (0x80)",
    "FLAG_REQUEST_ACCESSIBILITY_BUTTON (0x100)"
   ],
   "feedbackType": "[FEEDBACK_SPOKEN, FEEDBACK_VISUAL]"
  },
  {
   "id": "com.google.android.marvin.talkback/com.android.switchaccess.SwitchAccessService",
   "capabilities": [
    "CAPABILITY_CAN_RETRIEVE_WINDOW_CONTENT (0x1)",
    "CAPABILITY_CAN_REQUEST_FILTER_KEY_EVENTS (0x8)",
    "CAPABILITY_CAN_PERFORM_GESTURES (0x20)"
   ],
   "flags": [
    "DEFAULT (0x1)",
    "FLAG_REPORT_VIEW_IDS (0x10)",
    "FLAG_REQUEST_FILTER_KEY_EVENTS (0x20)",
    "FLAG_RETRIEVE_INTERACTIVE_WINDOWS (0x40)",
    "FLAG_ENABLE_ACCESSIBILITY_VOLUME (0x80)"
   ],
   "feedbackType": "[FEEDBACK_GENERIC]"
  },
  {
   "id": "com.google.android.apps.accessibility.auditor/.ScannerService",
   "capabilities": [
    "CAPABILITY_CAN_RETRIEVE_WINDOW_CONTENT (0x1)"
   ],
   "flags": [
    "FLAG_REPORT_VIEW_IDS (0x10)",
    "FLAG_RETRIEVE_INTERACTIVE_WINDOWS (0x40)"
   ],
   "feedbackType": "[FEEDBACK_GENERIC]"
  },
  {
   "id": "com.google.android.apps.accessibility.voiceaccess/.JustSpeakService",
   "capabilities": [
    "CAPABILITY_CAN_RETRIEVE_WINDOW_CONTENT (0x1)",
    "CAPABILITY_CAN_REQUEST_FILTER_KEY_EVENTS (0x8)"
   ],
   "flags": [
    "DEFAULT (0x1)",
    "FLAG_REPORT_VIEW_IDS (0x10)",
    "FLAG_REQUEST_FILTER_KEY_EVENTS (0x20)",
    "FLAG_RETRIEVE_INTERACTIVE_WINDOWS (0x40)"
   ],
   "feedbackType": "[FEEDBACK_GENERIC]"
  }
 ],
 "enabled": [
  "com.google.android.marvin.talkback/com.android.switchaccess.SwitchAccessService"
 ]
}```